File: datastructure using cpp/quicksort/quicksort.py
def partition(arr, l, h):
    pivot = arr[l]
    i = l
    j = h-1

    while i<j:
        while(pivot>=arr[i]):
            i += 1
        while(pivot<arr[j]):
            j-=1
        if i<j:
            arr[i],arr[j] = arr[j], arr[i]

    arr[l], arr[j] = arr[j], arr[l]
    return j

def quicksort(arr,l,h):
    if l<h:
        j = partition(arr,l,h)
        quicksort(arr, l, j-1)
        quicksort(arr, j+1, h)

arr = [10,16,8,12,15,6,3,9,5,8]
l = 0
h = len(arr)
quicksort(arr, l,h)
print(arr)

            

File: Sorting Algorithm/QuickSort.cpp
#include<iostream>
using namespace std;
int partion(int arr[],int l,int h){
	int pivot = arr[l];
	int i = l;
	int j = h-1;
	while(i<j){
		while(pivot>=arr[i])
			i ++;
		while(pivot<arr[j])
			j --;
		if (i<j){
			int temp = arr[i];
			arr[i] = arr[j];
			arr[j] = temp; 
		}
	}
	int temp = arr[l];
	arr[l] = arr[j];
	arr[j] = temp;
	
	return j;
	
}

void QuickSort(int arr[],int l,int h){
	if(l<h){
		int j = partion(arr,l,h);
		QuickSort(arr,l,j);
		QuickSort(arr,j+1,h);
	}
}

int main(){	
	int arr[]  = {4,2,52,6,44,99,2};
	int n = sizeof(arr)/sizeof(arr[0]);
	
	QuickSort(arr,0,n);
	for(int i =0; i<n; i++)
	{
		cout<<arr[i]<<endl;
	}
	return 0;
}
